files/image_config/interfaces/interfaces.j2

```diff
@@ -64,8 +64,11 @@ iface {{ port }} inet6 dhcp
 
 {% else %}
 {% if MGMT_INTERFACE %}
+{% for (name, prefix) in MGMT_INTERFACE|pfx_filter|unique(attribute=0) %}
```

Comment:
>unique [](http://example.com/codeflow?start=51&length=6)

Why one `for` has `unique`, but below `for` do not have? Just use one `for` block?

Reply:
if unique is not applied here, and each interface has an ipv4 and ipv6 address, then the interface will be repeated.
E.g., if we have
eth0|1.2.3.4
eth0|abcd::
eth1|2.3.4.5
eth1|cdef::

without unique, it would generate the following:
auto eth0
auto eth0
auto eth1
auto eth1

The loop below this does not need unique because ipv4 and ipv6 need to be iterated for the interface.